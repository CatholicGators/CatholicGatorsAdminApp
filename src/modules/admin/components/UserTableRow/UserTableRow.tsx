import React from "react"
import {
  TableRow,
  TableCell,
  Checkbox,
  Avatar,
  Switch,
  Theme,
  createStyles,
  withStyles
} from "@material-ui/core";

const styles = (theme: Theme) => createStyles({
  hiddenColxs: {
    [theme.breakpoints.down('xs')]: {
      display: 'none'
    }
  },
  hiddenColsm: {
    [theme.breakpoints.down('sm')]: {
      display: 'none'
    }
  },
  profilePicCol: {
    paddingRight: theme.spacing.unit * 2
  }
})

type Props = {
  user: any,
  classes: any,
  isSelected: boolean,
  handleSelect: (id) => any,
  handleApproveToggle: (user: any, checked: boolean) => any,
  handleAuthorizeToggle: (user: any, checked: boolean) => any
}

export const UserTableRow = (props: Props) => {
  const {
    user,
    classes,
    isSelected,
    handleSelect,
    handleApproveToggle,
    handleAuthorizeToggle
  } = props

  return (
    <TableRow
      hover
      aria-checked={isSelected}
      tabIndex={-1}
      key={user.id}
      selected={isSelected}
    >
      <TableCell padding="checkbox">
        <Checkbox
          onClick={() => handleSelect(user.id)}
          checked={isSelected}
        />
      </TableCell>
      <TableCell className={classes ? classes.profilePicCol : null}>
        <Avatar src={user.data.photoURL} />
      </TableCell>
      <TableCell className={classes ? classes.hiddenColsm : null}>{user.data.name}</TableCell>
      <TableCell>{user.data.email}</TableCell>
      <TableCell className={classes ? classes.hiddenColxs : null}>
        <Switch
          checked={user.data.isApproved}
          onChange={(_, checked) => handleApproveToggle(user, checked)}/>
      </TableCell>
      <TableCell className={classes ? classes.hiddenColxs : null}>
        <Switch
          disabled={!user.data.isApproved}
          checked={user.data.isAdmin}
          onChange={(_, checked) => handleAuthorizeToggle(user, checked)}
        />
      </TableCell>
    </TableRow>
  );
}

export default withStyles(styles)(UserTableRow)
