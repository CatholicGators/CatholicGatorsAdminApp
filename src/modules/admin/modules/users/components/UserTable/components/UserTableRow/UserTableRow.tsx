import React from "react"
import { connect } from "react-redux"
import {
    TableRow,
    TableCell,
    Checkbox,
    Avatar,
    Switch,
    Theme,
    createStyles,
    withStyles,
    IconButton,
} from "@material-ui/core"
import CheckIcon from "@material-ui/icons/Check"
import { updateUser } from "../../../../redux/actions/usersActions"

export const styles = (theme: Theme) =>
    createStyles({
        hiddenxs: {
            [theme.breakpoints.down("xs")]: {
                display: "none",
            },
        },
        hiddensm: {
            [theme.breakpoints.down("sm")]: {
                display: "none",
            },
        },
        avatarIconBtn: {
            padding: 0,
            [theme.breakpoints.up("sm")]: {
                display: "none",
            },
        },
        profilePicCol: {
            paddingRight: theme.spacing(2),
        },
    })

type Props = {
    user: any
    classes: any
    isSelected: boolean
    handleSelect: (id) => any
    updateUser: (user) => void
}

export class UserTableRow extends React.Component<Props> {
    handleApproveToggle(user, checked) {
        this.props.updateUser({
            ...user,
            isApproved: checked,
        })
    }

    handleAuthorizeToggle(user, checked) {
        this.props.updateUser({
            ...user,
            isAdmin: checked,
        })
    }

    render() {
        const { user, classes, isSelected, handleSelect } = this.props

        return (
            <TableRow
                hover
                aria-checked={isSelected}
                tabIndex={-1}
                key={user.id}
                selected={isSelected}
            >
                <TableCell
                    className={classes ? classes.hiddenxs : null}
                    padding="checkbox"
                >
                    <Checkbox
                        onClick={() => handleSelect(user.id)}
                        checked={isSelected}
                    />
                </TableCell>
                <TableCell className={classes ? classes.profilePicCol : null}>
                    <Avatar
                        className={classes ? classes.hiddenxs : null}
                        src={user.photoUrl}
                    />
                    <IconButton
                        className={classes ? classes.avatarIconBtn : null}
                        onClick={() => handleSelect(user.id)}
                    >
                        <Avatar src={isSelected ? null : user.photoUrl}>
                            {isSelected ? <CheckIcon /> : null}
                        </Avatar>
                    </IconButton>
                </TableCell>
                <TableCell className={classes ? classes.hiddensm : null}>
                    {user.name}
                </TableCell>
                <TableCell>{user.email}</TableCell>
                <TableCell>
                    <Switch
                        id="approve-switch"
                        checked={user.isApproved}
                        onChange={(_, checked) =>
                            this.handleApproveToggle(user, checked)
                        }
                    />
                </TableCell>
                <TableCell>
                    <Switch
                        id="authorize-switch"
                        disabled={!user.isApproved}
                        checked={user.isAdmin}
                        onChange={(_, checked) =>
                            this.handleAuthorizeToggle(user, checked)
                        }
                    />
                </TableCell>
            </TableRow>
        )
    }
}

const mapDispatchToProps = (dispatch) => ({
    updateUser: (user) => dispatch(updateUser(user)),
})

export default connect(
    null,
    mapDispatchToProps
)(withStyles(styles)(UserTableRow))
