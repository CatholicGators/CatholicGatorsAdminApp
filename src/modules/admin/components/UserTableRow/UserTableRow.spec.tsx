import React from 'react'
import { shallow } from 'enzyme';
import { UserTableRow } from './UserTableRow';

describe('UserTableRow', () => {
    let wrapper, props;

    beforeEach(() => {
        props = {
            user: {
                data: {
                    photoUrl: ''
                }
            },
            isSelected: true,
            handleSelect: jest.fn(),
            handleApproveToggle: jest.fn(),
            handleAuthorizeToggle: jest.fn()
        }
        wrapper = shallow(<UserTableRow {...props}/>)
    })

    it('matches snapshot', () => {
        expect(wrapper).toMatchSnapshot()
    })
})