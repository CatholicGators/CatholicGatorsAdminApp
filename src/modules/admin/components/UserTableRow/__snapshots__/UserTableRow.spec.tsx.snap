// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserTableRow matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UserTableRow
    handleApproveToggle={[MockFunction]}
    handleAuthorizeToggle={[MockFunction]}
    handleSelect={[MockFunction]}
    isSelected={true}
    user={
      Object {
        "data": Object {
          "photoUrl": "",
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "aria-checked": true,
      "children": Array [
        <WithStyles(TableCell)
          padding="checkbox"
        >
          <WithStyles(Checkbox)
            checked={true}
            onClick={[Function]}
          />
        </WithStyles(TableCell)>,
        <WithStyles(TableCell)
          className={null}
        >
          <WithStyles(Avatar) />
        </WithStyles(TableCell)>,
        <WithStyles(TableCell)
          className={null}
        />,
        <WithStyles(TableCell) />,
        <WithStyles(TableCell)
          className={null}
        >
          <WithStyles(Switch)
            onChange={[Function]}
          />
        </WithStyles(TableCell)>,
        <WithStyles(TableCell)
          className={null}
        >
          <WithStyles(Switch)
            disabled={true}
            onChange={[Function]}
          />
        </WithStyles(TableCell)>,
      ],
      "hover": true,
      "selected": true,
      "tabIndex": -1,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <WithStyles(Checkbox)
            checked={true}
            onClick={[Function]}
          />,
          "padding": "checkbox",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "checked": true,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <WithStyles(Avatar) />,
          "className": null,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "src": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": undefined,
          "className": null,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <WithStyles(Switch)
            onChange={[Function]}
          />,
          "className": null,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "checked": undefined,
            "onChange": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <WithStyles(Switch)
            disabled={true}
            onChange={[Function]}
          />,
          "className": null,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "checked": undefined,
            "disabled": true,
            "onChange": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "aria-checked": true,
        "children": Array [
          <WithStyles(TableCell)
            padding="checkbox"
          >
            <WithStyles(Checkbox)
              checked={true}
              onClick={[Function]}
            />
          </WithStyles(TableCell)>,
          <WithStyles(TableCell)
            className={null}
          >
            <WithStyles(Avatar) />
          </WithStyles(TableCell)>,
          <WithStyles(TableCell)
            className={null}
          />,
          <WithStyles(TableCell) />,
          <WithStyles(TableCell)
            className={null}
          >
            <WithStyles(Switch)
              onChange={[Function]}
            />
          </WithStyles(TableCell)>,
          <WithStyles(TableCell)
            className={null}
          >
            <WithStyles(Switch)
              disabled={true}
              onChange={[Function]}
            />
          </WithStyles(TableCell)>,
        ],
        "hover": true,
        "selected": true,
        "tabIndex": -1,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <WithStyles(Checkbox)
              checked={true}
              onClick={[Function]}
            />,
            "padding": "checkbox",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "checked": true,
              "onClick": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <WithStyles(Avatar) />,
            "className": null,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "src": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": undefined,
            "className": null,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <WithStyles(Switch)
              onChange={[Function]}
            />,
            "className": null,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "checked": undefined,
              "onChange": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <WithStyles(Switch)
              disabled={true}
              onChange={[Function]}
            />,
            "className": null,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "checked": undefined,
              "disabled": true,
              "onChange": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
